src/Symfony/Component/HttpClient/Tests/CurlHttpClientTest.php

```diff
@@ -42,4 +43,35 @@ public function testTimeoutIsNotAFatalError()
 
         parent::testTimeoutIsNotAFatalError();
     }
+
+    public function testOverridingRefererUsingCurlOptions()
+    {
+        $httpClient = $this->getHttpClient(__FUNCTION__);
+
+        $response = $httpClient->request('GET', 'http://localhost:8057/referer', [
+            'extra' => [
+                'curl' => [
+                    CURLOPT_REFERER => 'Banana',
+                ],
+            ],
+        ]);
+
+        $this->assertEquals('Banana', $response->getContent());
+    }
+
+    public function testOverridingHttpMethodUsingCurlOptions()
+    {
+        $httpClient = $this->getHttpClient(__FUNCTION__);
+
+        /** @var CurlResponse $response */
+        $response = $httpClient->request('POST', 'http://localhost:8057/method', [
+            'extra' => [
+                'curl' => [
+                    CURLOPT_HTTPGET => true,
```

Comment:
I think this example is not misleading (same for CURLOPT_REFERER btw):
we don't want to allow setting curlopts that already have a corresponding abstracted (aka portable) way IMHO.